src/messages/src_authority.rs

```diff
@@ -74,7 +91,7 @@ impl SrcAuthority {
     pub(crate) fn as_section_prefix(&self) -> Result<&Prefix> {
         match self {
             Self::Section { prefix, .. } => Ok(prefix),
-            Self::Node { .. } => Err(Error::InvalidSrcLocation),
+            _ => Err(Error::InvalidSrcLocation),
```

Comment:
prefer not to use wildcard match.
using `Self::Node { .. } | Self::BlsShare { .. }` ?